pydantic/_internal/_model_construction.py

```diff
@@ -430,6 +438,12 @@ def inspect_namespace(  # noqa C901
             and ann_type not in all_ignored_types
             and getattr(ann_type, '__module__', None) != 'functools'
         ):
+            if isinstance(ann_type, str):
+                # Walking up the frames to get the module namespace where the model is defined
+                # (as the model class wasn't created yet, we unfortunately can't use `cls.__module__`):
+                module = inspect.getmodule(sys._getframe(2))
+                if module is not None:
+                    ann_type = eval_type_backport(ForwardRef(ann_type), globalns=module.__dict__)
```

Comment:
```suggestion
                frame = sys._getframe(2)
                if frame is not None:
                    ann_type = eval_type_backport(ForwardRef(ann_type), globalns=frame.f_globals, localns=frame.f_locals)
```

Reply:
Agreed with a movement towards this approach, but should we just use https://github.com/pydantic/pydantic/blob/72124848b53bfc648af2d3f28ba54ff392c623ec/pydantic/_internal/_typing_extra.py#L188?

If we go with that, then you can ignore all of my above comments, bc I think this would address them.

Reply:
`parent_frame_namespace` only considers locals so I went with the `frame` locals and globals

Reply:
Sounds good to me. Thanks for following up!